### ISSUE TITLE
vmImage 'macOS-10.14': support for 'iPhone 5s' running iOS 12

### ISSUE THREAD

**Issue by jayaddison** [actions/virtual-environments#717]
**Describe the bug**
The [html2canvas](https://github.com/niklasvh/html2canvas) project uses [Azure Pipelines](https://github.com/niklasvh/html2canvas/blob/ae5f866b37ba9903614965e0aef4366feabd32a6/azure-pipelines.yml) to build and test across multiple operating systems and browsers (e.g. [Safari on iPhone 5s running iOS 12](https://github.com/niklasvh/html2canvas/blob/ae5f866b37ba9903614965e0aef4366feabd32a6/azure-pipelines.yml#L87-L92)).

As announced in a recent Azure DevOps [blog post](https://devblogs.microsoft.com/devops/removing-older-images-in-azure-pipelines-hosted-pools/) (and [duplicate post](https://devblogs.microsoft.com/devops/azure-pipelines-hosted-pools-updates/)?), the vmImage `macOS-10.13` is now unavailable and https://github.com/niklasvh/html2canvas/pull/2204 is an attempt to upgrade to a more recent vmImage.

The [documentation for `macOS-10.14`](https://github.com/microsoft/azure-pipelines-image-generation/blob/f44d37925cb9ad4df7d11d3d25578d85c1370a81/images/macos/macos-10.14-Readme.md#installed-simulators) seems to indicate that `iPhone 5s` on `iOS 12` should still be a supported simulator.

During builds of `html2canvas` against `macOS-10.14`, `iPhone 5s` is not found as a supported device for `iOS 12`:
1. https://github.com/niklasvh/html2canvas/pull/2204/commits/40d3689c0e1c91aa0bf28f09d965cfa7395f1286 : [log ref](https://dev.azure.com/niklasvh/html2canvas/_build/results?buildId=258&view=logs&j=3b7a686e-0620-5f5b-9d43-3cfc2ab5c523&t=86b8de10-f64e-5b14-e9fb-030c6583e882&l=22) (`ERROR [launcher:MobileSafari]: No device found for sdk 12.1 with name iPhone 5s`)
1. https://github.com/niklasvh/html2canvas/pull/2204/commits/409e24cac005cce0a1f42f95d4bfd4ef4a352145 : [log ref](https://dev.azure.com/niklasvh/html2canvas/_build/results?buildId=265&view=logs&j=3b7a686e-0620-5f5b-9d43-3cfc2ab5c523&t=86b8de10-f64e-5b14-e9fb-030c6583e882&l=97) (`ERROR [launcher:MobileSafari]: No device found for sdk 12.1 with name iPhone 5s
`)

**Question**
Is vmImage `macOS-10.14` still supported and if so does it contain simulator support for `iPhone 5s` on `iOS 12`?

**Area for Triage**: Apple

**Question, Bug, or Feature?**: Question

**Virtual environments affected**
- [x] macOS 10.14
- [ ] macOS 10.15

**Comment by maxim-lobanov** [comment]
Hello @jayaddison!
What version of Xcode do you use?
I have looked at azure-pipelines yml files in [html2canvas](https://github.com/niklasvh/html2canvas) and it looks like the default one is used.
Default `Xcode` is different for `macos-10.13` and `macos-10.14` images. `Xcode 10.1` is default one on `macos-10.13` and `Xcode 11.3.1` is default one on `macos-10.14`.
So based on documentation that you have shared, 12.1 sdk is available only for Xcode 10.1 (second column)
![image](https://user-images.githubusercontent.com/16715858/79239754-99851600-7e79-11ea-9816-1d10a70ec880.png)

Solution:
We need to switch back to `Xcode 10.1`(it is still available on `macos-10.14` image).
You should add the following step to your pipeline:
```
- script: sudo xcode-select -s "/Applications/Xcode_10.1.app"
  displayName: 'Switch Xcode'
```

Could you please try and let me know if it works for you?

**Comment by jayaddison** [comment]
>  We need to switch back to Xcode 10.1(it is still available on macos-10.14 image).
> ...
> Could you please try and let me know if it works for you?

Thanks for the swift response @maxim-lobanov!

After some typical battles with YAML, this should be [running in a moment](https://dev.azure.com/niklasvh/html2canvas/_build/results?buildId=270&view=logs&j=3b7a686e-0620-5f5b-9d43-3cfc2ab5c523&t=9a87b47e-d0f9-5163-11e3-c17148b46c5d).

**Comment by jayaddison** [comment]
The build for `iOS Simulator Safari 12` has [succeeded](https://dev.azure.com/niklasvh/html2canvas/_build/results?buildId=271&view=logs&j=3b7a686e-0620-5f5b-9d43-3cfc2ab5c523&t=9a87b47e-d0f9-5163-11e3-c17148b46c5d) after the suggested changes.  Thanks @maxim-lobanov! :tada:
